packages/frontend/src/routes/SignUp/SignUp.jsx

```diff
@@ -14,25 +19,34 @@ export default function SignUp() {
   const [password, setPassword] = useState("");
   const [confirmPassword, setConfirmPassword] = useState("");
   const [passwordError, setPasswordError] = useState(false);
+  const [loading, setLoading] = useState(false);
+  const { setCurrentUser } = useContext(AuthContext);
+  const navigate = useNavigate();
+
+  async function handleSubmit(e) {
+    e.preventDefault();
 
-  const handleSubmit = (event) => {
-    event.preventDefault();
     if (password !== confirmPassword) {
       setPasswordError(true);
-    } else {
-      setPasswordError(false);
-      const formData = { firstName, lastName, email, password };
-      console.log(formData);
-      // axios
-      //   .post("api/endpoint", formData)
-      //   .then((response) => {
-      //     console.log(response);
-      //   })
-      //   .catch((error) => {
-      //     console.log(error);
-      //   });
+      return;
+    }
+    setPasswordError(false);
+    setLoading(true);
+
+    try {
+      await createUserWithEmailAndPassword(auth, email, password).then(
+        (userCredential) => {
+          const user = userCredential.user;
+          console.log("User SIGNED UP ith the email", auth?.currentUser?.email);
+          setCurrentUser(user);
+          navigate("/");
```

Comment:
can we navigate to /login after signup, or does it automatically log user in after sign up?

Reply:
Yea it is possible to redirect the user to login page after signup and make them log in again. Would just need to not set the user state and set logged in to false.
But isn't it better if a user automatically signs in after signing up? Would be inconvenient for them to log in again.

Reply:
oh nice, so if it can auto login user after signup then that's better